self-core-impl/src/test/java/com/selfxdsd/core/mock/InMemoryContracts.java

```diff
@@ -22,43 +22,392 @@
+         *
+         * @param contract Contract to be added.
+         */
+        void addContract(final Contract contract) {
+            ownedContracts.add(contract);
+        }
+
+        @Override
+        public int projectId() {
+            return delegate.projectId();
+        }
+
+        @Override
+        public ProjectManager projectManager() {
+            return delegate.projectManager();
+        }
+
+        @Override
+        public Repo repo() {
+            return delegate.repo();
+        }
+
+        @Override
+        public Contracts contracts() {
+            return new Contracts() {
+                @Override
+                public Contracts ofProject(final int projectId) {
+                    return rootContracts.ofProject(projectId);
+                }
+
+                @Override
+                public Iterator<Contract> iterator() {
+                    return ownedContracts.iterator();
+                }
+            };
+        }
+
+        @Override
+        public Repo deactivate() {
+            return null;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hashCode(delegate.projectId());
+        }
+
+        @Override
+        public boolean equals(final Object obj) {
+            if (!(obj instanceof Project)) {
+                return false;
+            }
+            final Project other = (Project) obj;
+            return other.projectId() == delegate.projectId();
+        }
+    }
+
+    /**
+     * In memory implementation of a {@link Contributor}.
+     * <br>
+     * Aids {@link InMemoryContracts}
+     * <br>
+     * Has capability to link a contract.
+     * See: {@link ContributorInternal#addContract(Contract)}.
+     */
+    private static final class ContributorInternal implements Contributor {
+
+        /**
+         * Contributors's id.
+         */
+        private final int id;
+        /**
+         * Global contracts. Access to {@link InMemoryContracts }
+         * Grants access to any arbitrary project's contracts using
+         * {@link Contracts#ofProject(int) from any point of the object graph.
+         */
+        private final Contracts rootContracts;
+        /**
+         * Contracts associated to this contributor.
+         */
+        private final Set<Contract> ownedContracts = new HashSet<>();
+
+        /**
+         * Constructor.
+         *
+         * @param id Project id
+         * @param rootContracts Global contracts.
+         *                      Access to {@link InMemoryContracts }
+         */
+        private ContributorInternal(final int id,
+                                    final Contracts rootContracts) {
+            this.rootContracts = rootContracts;
+            this.id = id;
+        }
+
+        @Override
+        public int contributorId() {
+            return id;
+        }
+
+        @Override
+        public Contracts contracts() {
+            return new Contracts() {
```

Comment:
@criske THis method should throw UnsupportedOperationException at the moment, because Contracts.ofContributor(...) does not exist yet.

Reply:
but then it will fail Contributor#contracts() tests